ddm_test_validate.py

```diff
@@ -0,0 +1,214 @@
+            log_df.to_csv(
+                path_or_buf=fname, mode="a", header=False, index=False
+            )
+        else:
+            log_df.to_csv(path_or_buf=fname, mode="w", header=True, index=False)
+
+def test_random_policy(
+    num_episodes: int = 100,
+    num_iterations: int = 250,
+    log_iterations: bool= True,
+    sim: Simulator = None,
+):
+    """Test a policy using random actions over a fixed number of episodes
+
+    Parameters
+    ----------
+    num_episodes : int, optional
+        number of iterations to run, by default 10
+    """
+    for episode in range(num_episodes):
+        iteration = 0
+        terminal = False
+        sim.episode_start()
+        ddm_state = sim.get_state()
+        sim_state = sim.get_sim_state()
+        # it is important to know initial actions for evolution of the dynamics
+        # action = random_action()
+        
+        action = sim.get_init_sim_actions()
+        if log_iterations:
+            sim.log_iterations(ddm_state, action, sim.log_file, episode, iteration)
+            sim.log_iterations(sim_state, action, sim.log_file2, episode, iteration)
+        while not terminal:
+            action = random_policy(ddm_state)
+            # sim iteration
+            sim.episode_step(action)
+            sim.sim_physics.episode_step(action)
+            ddm_state = sim.get_state()
+            sim_state = sim.get_sim_state()
+            iteration += 1
+            if log_iterations:
+                sim.log_iterations(ddm_state, action, sim.log_file, episode, iteration)
+                sim.log_iterations(sim_state, action, sim.log_file2, episode, iteration)
+            print(f"Running iteration #{iteration} for episode #{episode}")
+            print(f"Observations for Sim: {sim_state}")
+            print(f"Observations for Data: {ddm_state}")
+            terminal = iteration >= num_iterations
+
+    return sim
+    
+
+
+@hydra.main(config_path="conf", config_name="config")
+def main(cfg: DictConfig):
+
+    save_path = cfg["model"]["saver"][0]["filename"]
+    if cfg["data"]["full_or_relative"] == "relative":
+        save_path = os.path.join(dir_path, save_path)
+    model_name = cfg["model"]["name"]
+    states = cfg["simulator"]["states"]
+    actions = cfg["simulator"]["actions"]
+    configs = cfg["simulator"]["configs"]
+    policy = cfg["simulator"]["policy"]
+    logflag = cfg["simulator"]["logging"]
+    scale_data = cfg["model"]["build_params"][7]["scale_data"]
+
+    logger.info(f"Training with a new {policy} policy")
+
+    ddModel = available_models[model_name]
+    model = ddModel()
+
+    model.build_model(model_type=model_name)
+    model.load_model(filename=save_path, scale_data=scale_data)
+
+    # Grab standardized way to interact with sim API
+    sim = Simulator(model, states, actions, configs, logflag)
+    
+    test_random_policy(100, 250,logflag=="enable", sim)
+    
+    return sim
+
+        
+
+if __name__ == "__main__":
+
+    main()
```

Comment:
hmm, it seems to me this entire script is basically the same as having the `policy=random` flag in `ddm_predictor`, although that script was defined for inferencing the ddm, not the original simulator. That said, it could be used for that purpose?

Reply:
The intent is to allow multiple types of tests. Random policy is one of them. Coasting might be another. There might be other case specific tests for specific use case that might require to be scripted specifically. So, perhaps we could retain policies.py, and use that as a place for users to define their own test cases. (Initially, I wasn't in favor of policies.py. It seemed odd, but then when I see how much testing simulators is important, and how much in practice users ignore testing simulators (or ddms), it seems reasonable to keep it, now. We can always change it, or change its form once the software design for DDM firms up. I think that is going to be some incremental process.